### PR TITLE
Fix API Product invoke from swagger console

### PR DESCRIPTION
**Pull request by uvindra**
Enable invoking API products via Swagger console. This includes a further generalization of the new swagger parsing code.

Product_Console

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/SwaggerData.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    private String description;
+    private String version;
+    private String contactName;
+    private String contactEmail;
+    private String transportType;
+    private String security;
+    private String apiLevelPolicy;
+    private Set<Resource> resources = new HashSet<>();
+    private Set<Scope> scopes = new HashSet<>();
+
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/SwaggerData.java]: there are multiple empty lines. Hope we can remove them.

Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/SwaggerData.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiProductsApiServiceImpl.java [hunk 2]
```diff
@@ -139,12 +139,18 @@ public Response apiProductsApiProductIdSubscriptionPoliciesGet(String apiProduct
         return null;
     }
 
-    @Override public Response apiProductsApiProductIdSwaggerGet(String apiProductId, String ifNoneMatch,
-            String xWSO2Tenant, MessageContext messageContext) {
+    @Override public Response apiProductsApiProductIdSwaggerGet(String apiProductId, String labelName,
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiProductsApiServiceImpl.java]: i think @Override should be in a new line.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiProductsApiServiceImpl.java [hunk 3]
```diff
@@ -153,10 +159,22 @@ public Response apiProductsApiProductIdSubscriptionPoliciesGet(String apiProduct
             if (product == null) {
                 RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API_PRODUCT, apiProductId, log);
             }
+
+            //gets the first available environment if neither label nor environment is not provided
+            if (org.apache.commons.lang3.StringUtils.isEmpty(labelName) && org.apache.commons.lang3.StringUtils.isEmpty(environmentName)) {
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiProductsApiServiceImpl.java]: seems maximum length is exceeded.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java [hunk 4]
```diff
@@ -1685,7 +1687,8 @@ public Response apisApiIdThumbnailPost(String apiId, InputStream fileInputStream
             String apiSwaggerDefinition = apiProvider.getOpenAPIDefinition(api.getId());
             if (!StringUtils.isEmpty(apiSwaggerDefinition)) {
                 APIDefinition apiDefinitionFromOpenAPISpec = new APIDefinitionFromOpenAPISpec();
-                Set<URITemplate> uriTemplates = apiDefinitionFromOpenAPISpec.getURITemplates(api, apiSwaggerDefinition);
+                SwaggerData swaggerData = new SwaggerData(api);
+                Set<URITemplate> uriTemplates = apiDefinitionFromOpenAPISpec.getURITemplates(swaggerData, apiSwaggerDefinition);
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: seems length is exceeded.

**Pull request merged**